### PR TITLE
Fix GetPathValue function

### LINKED ISSUE #1881: Unable to save settings
**Describe the bug**
Settings can't be saved on any page that has a save button.

**To Reproduce**
1. Go to any page that has a save button eg Dashboard - Playback.
2. Click on Save.
3. Witness a never ending loading spinner.
4. Settings are never saved.

**Expected behavior**
Settings being saved.

**Logs**
```[ERR] Error processing request
System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at Emby.Server.Implementations.AppBase.BaseConfigurationManager.GetConfigurationType(String key)
   at MediaBrowser.Api.ConfigurationService.Post(UpdateNamedConfiguration request)
   at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task)
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken)
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
```

**System (please complete the following information):**
 - OS: [Windows but probably any]
 - Browser: [Any]
 - Jellyfin Version: [e.g. 10.4.0 Nightly 10/10]
 - Reverse proxy: [No]

**Note by @JustAMan**: edited logs for clarity

### LINKED ISSUE #2009: After 10.4.1 can't add custom images
**Describe the bug**
I have 10.4.1 and using baseurl = /jellyfin and also nginx to redirect to www.domain.com/jellyfin from the internet.

Since the upgrade I can't add custom images (from disk) to any tvshows (I assume is for movies as well). For example if JF can't find a thumb of a tvshow and I would like to add one from disk JF it's just spinning and doesn't complete the process: (tvshow / edit images ). 

If I remove the baseurl it works and I can add the custom images.

**To Reproduce**
Go to a TV-Show.
Click on edit images
Click on the "+" and chose a image file from disk

**Expected behavior**
Image from disk is picked up from disk and upload it into JF.
**Logs**
```
[2019-11-13 13:28:21.883 +00:00] [ERR] Error processing request
System.ArgumentException: Requested value 'Images' was not found.
   at System.Enum.TryParseEnum(Type enumType, String value, Boolean ignoreCase, EnumResult& parseResult)
   at System.Enum.Parse(Type enumType, String value, Boolean ignoreCase)
   at MediaBrowser.Api.Images.ImageService.Post(PostItemImage request)
   at Emby.Server.Implementations.Services.ServiceExecGeneral.Execute(Type serviceType, IRequest request, Object instance, Object requestDto, String requestName)
   at Emby.Server.Implementations.Services.ServiceController.Execute(HttpListenerHost httpHost, Object requestDto, IRequest req)
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken)
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
```

**System (please complete the following information):**
 - OS: Docker
 - Jellyfin Version: 10.4.1
 - Reverse proxy: nginx

**Additional context**
If I remove the baseurl it works and I can add the custom images. This was working on 10.4.0.

### PR DESCRIPTION
**Pull request by Bond-009**
**Issues**
Fixes #1881
Fixes #2009

### CODE REVIEW

File: MediaBrowser.Api/BaseApiService.cs [hunk 1]
```diff
@@ -299,8 +299,9 @@ protected string GetPathValue(int index)
...
 
             // backwards compatibility
-            if (string.Equals(first, "mediabrowser", StringComparison.OrdinalIgnoreCase) ||
-                string.Equals(first, "emby", StringComparison.OrdinalIgnoreCase))
+            if (string.Equals(first, "mediabrowser", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(first, "emby", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(first, ApiEntryPoint.Instance.ConfigurationManager.Configuration.BaseUrl))
             {
                 index++;
             }
```
Comment (JustAMan) [inline MediaBrowser.Api/BaseApiService.cs]: Won't this break for `/baseurl/emby` now?.. maybe do something like (untested)
```cs
int urlIndex = 0;
var urlPart = pathInfo[urlIndex];

while (string.Equals(urlPart, "mediabrowser", StringComparison.OrdinalIgnoreCase) // backwards compatibility
                || string.Equals(urlPart, "emby", StringComparison.OrdinalIgnoreCase) // backwards compatibility
                || string.Equals(urlPart, ApiEntryPoint.Instance.ConfigurationManager.Configuration.BaseUrl))
{
    index++;
    urlIndex++;
    urlPart = pathInfo[urlIndex];
}
```
?

Comment (Bond-009) [reply MediaBrowser.Api/BaseApiService.cs]: Good catch.

File: MediaBrowser.Api/BaseApiService.cs [hunk 2]
```diff
@@ -298,11 +297,24 @@ protected string GetPathValue(int index)
...
+            }
+            else if (string.Equals(first, baseUrl))
             {
                 index++;
+                var second = pathInfo[1];
+                if (string.Equals(second, "mediabrowser", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(second, "emby", StringComparison.OrdinalIgnoreCase))
+                {
+                    index++;
+                }
```
Comment (JustAMan) [inline MediaBrowser.Api/BaseApiService.cs]: this looks to be copied twice, can we extract it to a common method, probably placed in `ApiEntryPoint.Instance.ConfigurationManager.Configuration`?

Comment (Bond-009) [reply MediaBrowser.Api/BaseApiService.cs]: I don't think putting this method in the configuration class is a good idea. These 2 classes share a lot of code, and IMO de-duplicating this is out of the scope of this PR.

**Comment by joshuaboniface** [comment]
Backported into 10.4.2.

**Comment by joshuaboniface** [comment]
This might be broken on Master and definitely was on release.

By removing `using MediaBrowser.Model.Extensions;` in `Emby.Server.Implementations/HttpServer/HttpListenerHost.cs`, the following error was thrown:

```
  MediaBrowser.Providers -> /jellyfin/MediaBrowser.Providers/bin/Release/netstandard2.0/MediaBrowser.Providers.dll
  Emby.Dlna -> /jellyfin/Emby.Dlna/bin/Release/netstandard2.0/Emby.Dlna.dll                            
HttpServer/HttpListenerHost.cs(270,23): error CS1501: No overload for method 'Replace' takes 3 arguments [/jellyfin/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
HttpServer/HttpListenerHost.cs(271,23): error CS1501: No overload for method 'Replace' takes 3 arguments [/jellyfin/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
make[1]: *** [debian/rules:42: override_dh_auto_build] Error 1
```

**Pull request merged**